src/applications/disability-benefits/new-686/config/chapters/report-divorce/former-spouse-information/formerSpouseInformation.js

```diff
@@ -0,0 +1,81 @@
+import currentOrPastDateUI from 'platform/forms-system/src/js/definitions/currentOrPastDate';
+import { validateName } from '../../../utilities';
+import { genericSchemas } from '../../../generic-schema';
+
+const {
+  fullName: formerSpouseName,
+  date: dateOfDivorce,
+  genericLocation: locationOfDivorce,
+} = genericSchemas;
+
+export const schema = {
+  type: 'object',
+  required: [
+    'formerSpouseName',
+    'dateOfDivorce',
+    'locationOfDivorce',
+    'isMarriageAnnulledOrVoid',
+  ],
+  properties: {
+    formerSpouseName,
+    dateOfDivorce,
+    locationOfDivorce,
+    isMarriageAnnulledOrVoid: { type: 'boolean' },
+    explanationOfAnnullmentOrVoid: {
+      type: 'string',
+      maxLength: 500,
+      pattern: '^(?!\\s)(?!.*?\\s{2,})[^<>%$#@!^&*]+$',
+    },
+  },
+};
+
+export const uiSchema = {
+  formerSpouseName: {
+    'ui:validations': [validateName],
+    first: {
+      'ui:title': 'Former spouse’s first name',
+      'ui:errorMessages': { required: 'Please enter a first name' },
+    },
+    middle: { 'ui:title': 'Former spouse’s middle name' },
+    last: {
+      'ui:title': 'Former spouse’s last name',
+      'ui:errorMessages': { required: 'Please enter a last name' },
+    },
+    suffix: {
+      'ui:title': 'Former spouse’s suffix',
+      'ui:options': {
+        widgetClassNames: 'form-select-medium',
+        hideIf: () => true,
+      },
+    },
+  },
+  dateOfDivorce: currentOrPastDateUI('Date of divorce'),
+  locationOfDivorce: {
+    'ui:title': 'Where did this marriage end?',
+    state: {
+      'ui:title': 'State (or country if outside USA)',
+      'ui:errorMessages': {
+        required: 'Please enter a state, or country if outside of USA',
+      },
+    },
+    city: {
+      'ui:title': 'City or county',
+      'ui:errorMessages': {
+        required: 'Please enter a city or county',
+      },
+    },
```

Comment:
Shouldn't city/county come before state/country?

Reply:
I know it's sort of weird, but following our mockups again [here](https://xd.adobe.com/view/0f7759a1-b990-4d71-50c9-138d9e593fd0-7946/screen/dbaba763-e3bd-4633-bc8c-24a5b3ca36cf/Apply-Dependents-Report-Divorce-1A)

Reply:
Ahh, maybe it's because the selected state could change the list of cities... at least if autocomplete was set up.